perllib/FixMyStreet/App/Controller/Admin/Waste.pm

```diff
@@ -0,0 +1,112 @@
+package FixMyStreet::App::Controller::Admin::Waste;
+
+use JSON::MaybeXS;
+use Moose;
+use Try::Tiny;
+use namespace::autoclean;
+
+BEGIN { extends 'Catalyst::Controller'; }
+
+=head1 NAME
+
+FixMyStreet::App::Controller::Admin::Waste - Catalyst Controller
+
+=head1 DESCRIPTION
+
+Admin pages for configuring WasteWorks parameters
+
+=head1 METHODS
+
+=cut
+
+sub index : Path : Args(0) {
+    my ( $self, $c ) = @_;
+
+    my $user = $c->user;
+
+    if ($user->is_superuser) {
+        $c->forward('fetch_wasteworks_bodies');
+    } elsif ( $user->from_body ) {
+        $c->forward('load_wasteworks_body', [ $user->from_body->id ]);
+        $c->res->redirect( $c->uri_for_action( 'admin/waste/edit', $c->stash->{body}->id ) );
+    } else {
+        $c->detach( '/page_error_404_not_found', [] );
+    }
+}
+
+sub edit : Path : Args(1) {
+    my ( $self, $c, $body_id ) = @_;
+
+    foreach (qw(status_message)) {
+        $c->stash->{$_} = $c->flash->{$_} if $c->flash->{$_};
+    }
+
+    $c->forward('load_wasteworks_body', [ $body_id ]);
+    $c->forward('stash_body_config_json');
+    $c->forward('/auth/get_csrf_token');
+
+    if ($c->req->method eq 'POST') {
+        $c->forward('/auth/check_csrf_token');
+
+        my $new_cfg;
+        try {
+            $new_cfg = JSON->new->utf8(1)->allow_nonref->decode($c->get_param("body_config"));
+        } catch {
+            $c->stash->{errors} ||= {};
+            my $e = $_;
+            $e =~ s/ at \/.*$//; # trim the filename/lineno
+            $c->stash->{errors}->{body_config} = sprintf(_("Not a valid JSON string: %s"), $e);
```

Comment:
If I recall correctly, the _() method is related to translation/localisation...?

Reply:
It is indeed. This part of the codebase is technically not Peterborough or even UK specific so it doesn't hurt to get it ready for potential future translation.